### PR TITLE
Enable build for Python <3.11

### PR DESCRIPTION
**Pull request by borchero**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by borchero** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -16,10 +16,9 @@ source:
     folder: pyston/LuaJIT/
 
 build:
-  number: 2
+  number: 3
 
   skip: True  # [not linux]
```
Comment (xhochy) [inline recipe/meta.yaml]:
```suggestion
  skip: True  # [not linux or (python_impl == "pypy") or py>311]
```

**Comment by borchero** [comment]
@conda-forge-admin, please rerender

File: .ci_support/linux_64_python3.10.____cpythonpython_implpypy.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+docker_image:
+- quay.io/condaforge/linux-anvil-cos7-x86_64
+pin_run_as_build:
+  python:
+    min_pin: x.x
+    max_pin: x.x
+python:
+- 3.10.* *_cpython
+python_impl:
+- pypy
```
Comment (xhochy) [inline .ci_support/linux_64_python3.10.____cpythonpython_implpypy.yaml]: 🤷 Something went wrong here.

**Comment by borchero** [comment]
@conda-forge/pyston_lite This is ready for review/merge.

**Pull request merged**